### PR TITLE
Rehaul 'how to integrate your library' docs

### PR DESCRIPTION
**Pull request by Wauplin**
I first started to work on this PR to document how to integrate a library with the Hub (i.e. how to open the PR on huggingface.js). In the end, I've updated more than that to focus the page on "how to implement the integration?" + "how to register your library?". 

I feel that some sections are outdated, typically the "how to login" (out of scope for a library integration?) and "how to add inference API" (we don't want to add new libs to api-inference-community, right?) so I removed them. Let me know if you feel strongly against it.

Once merged, I will open a PR to delete https://github.com/huggingface/huggingface.js/blob/main/.github/pull_request_template/new_library.md which is currently useless (not discoverable :confused:)

---

**For reviewers:** https://moon-ci-docs.huggingface.co/docs/hub/pr_1442/en/models-adding-libraries

### CODE REVIEW

File: docs/hub/models-adding-libraries.md [hunk 1]
```diff
@@ -6,116 +6,93 @@ Integrating the Hub with your library provides many benefits, including:
...
+Authentication is optional when downloading files from public repos on the Hub.
 
-The commit hash must be a full-length hash instead of the shorter 7-character commit hash:
+### Download files from the Hub
 
-```python
->>> from huggingface_hub import hf_hub_download
->>> hf_hub_download(repo_id="lysandre/arxiv-nlp", filename="config.json", revision="877b84a8f93f2d619faa2a6e514a32beef88ab0a")
-```
+Integrations allows users to download a model from the Hub and instantiate it directly from yours library. This is often made possible by providing a method (usually called `from_pretrained` or `load_from_hf`) that has be specific to your library. To instantiate a model from the Hub, your library has to:
```
Comment (osanseviero) [inline docs/hub/models-adding-libraries.md:61]:
```suggestion
Integrations allow users to download a model from the Hub and instantiate it directly from your library. This is often made possible by providing a method (usually called `from_pretrained` or `load_from_hf`) that has to be specific to your library. To instantiate a model from the Hub, your library has to:
```

File: docs/hub/models-adding-libraries.md [hunk 2]
```diff
@@ -6,116 +6,93 @@ Integrating the Hub with your library provides many benefits, including:
...
+- download files from the Hub. This is what we will discuss now.
+- instantiate the Python model from these files.
 
-Use the `cache_dir` parameter to change where a file is stored:
+Use the [`hf_hub_download`](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/file_download#huggingface_hub.hf_hub_download) method to download files from a repository on the Hub. Downloaded files are stored in the cache: `~/.cache/huggingface/hub`. Users won't have to re-download the file the next time they use it, which saves a lot of time for large files. Furthermore, if the repository is updated with a new version of the file, `huggingface_hub` will automatically download the latest version and store it in the cache. Users don't have to worry about updating their files manually.
```
Comment (osanseviero) [inline docs/hub/models-adding-libraries.md:65]:
```suggestion
Use the [`hf_hub_download`](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/file_download#huggingface_hub.hf_hub_download) method to download files from a repository on the Hub. Downloaded files are stored in the cache: `~/.cache/huggingface/hub`. Users won't have to re-download the file the next time they use it, which saves a lot of time for large files. Furthermore, if the repository is updated with a new version of the file, `huggingface_hub` will automatically download the latest version and store it in the cache. Users don't have to worry about updating their files manually.
```

File: docs/hub/models-adding-libraries.md [hunk 3]
```diff
@@ -145,51 +120,74 @@ For example:
...
 
-Lastly, it is important to add a model card so users understand how to use your model. See [here](./model-cards) for more details about how to create a model card.
+If your library allows pushing a model to the Hub, it is recommended to generate a minimal model card with prefilled metadata (typically `library_name`, `pipeline_tag` or `tags`) and information on how the model has been trained. This will help having a standardized description for all models built with your library.
 
-## Set up the Inference API
+## Register your library
 
-Our Inference API powers models uploaded to the Hub through your library.
+Well done! You should now have a library able to load a model from the Hub and eventually push new models. The next step is to make sure that your models on the Hub are well documented and integrated with the platform. To do so, libraries can be registered on the Hub, which comes with a few benefits for the users:
```
Comment (osanseviero) [inline docs/hub/models-adding-libraries.md:145]:
```suggestion
Well done! You should now have a library able to load a model from the Hub and eventually push new models. The next step is to make sure that your models on the Hub are well-documented and integrated with the platform. To do so, libraries can be registered on the Hub, which comes with a few benefits for the users:
```

**Review by osanseviero — COMMENTED** [review]
Looking very nice!

File: docs/hub/models-adding-libraries.md [hunk 4]
```diff
@@ -6,116 +6,93 @@ Integrating the Hub with your library provides many benefits, including:
...
-   ```
+```
+npm add @huggingface/hub
+```
 
-   `notebook_login` will launch a widget in your notebook from which you can enter your Hugging Face credentials.
+</Tip>
 
-## Download files from the Hub
+Once they have successfully installed the `huggingface_hub` library, users will need to authenticate. The easiest way to authenticate is to save the token on the machine. You can do that from the terminal using the `login()` command:
```
[the remark this answers is not in the document]
Comment (Wauplin) [reply docs/hub/models-adding-libraries.md]: Agree on been consistent here. In the guide I tried to use "you" to refer to the library maintainer and "they" for the end users. But yeah, this section was really inconsistent. Addressed in https://github.com/huggingface/hub-docs/pull/1442/commits/9d484b13020bb738c331aba2a51a62e7fd2efb40.

File: docs/hub/models-adding-libraries.md (hunk 2 above)
Comment (Wauplin) [reply docs/hub/models-adding-libraries.md:65]: Missing suggestion? :smile:

**Review by Vaibhavs10 — APPROVED** [review]
Niceee! 🔥 - Read through the PR didn't find any grammatical errors and such. Love the overall flow of the doc as well.

File: docs/hub/models-adding-libraries.md [hunk 5]
```diff
@@ -145,51 +120,69 @@ For example:
...
-    * Add your model and task to the `tests/test_api.py` file. For example, if you have a text generation model:
+Before opening the PR, make sure that at least one model is referenced on https://huggingface.co/models?other=my-library-name. If not, the model card metadata of the relevant models must be updated with `library_name: my-library-name` (see [example](https://huggingface.co/google/gemma-scope/blob/main/README.md?code=true#L3)). If you are not the owner of the models on the Hub, please open PRs (see [example](https://huggingface.co/MCG-NJU/VFIMamba/discussions/1)).
 
-       ```python
-       TESTABLE_MODELS: Dict[str,str] = {
-           "text-generation": "my-gpt2-model"
-       }
-       ```
-3. Finally, run the following test to ensure everything works as expected:
+Here is a minimal [example](https://github.com/huggingface/huggingface.js/pull/561/files) adding integration for Grok-1.
```
Comment (osanseviero) [inline docs/hub/models-adding-libraries.md:163]: Should we have an example including a snippet and a simpler download rule?

**Review by osanseviero — APPROVED** [review]
Nice! :fire:

File: docs/hub/models-adding-libraries.md (hunk 5 above)
Comment (Vaibhavs10) [reply docs/hub/models-adding-libraries.md:163]: What about this one? https://github.com/huggingface/huggingface.js/pull/885/files - It looks quite clean and has download counts and snippets both.

Comment (Vaibhavs10) [inline docs/hub/models-adding-libraries.md:163]:
```suggestion
Here is a minimal [example](https://github.com/huggingface/huggingface.js/pull/885/files) adding integration for VFIMamba.
```

Comment (Vaibhavs10) [reply docs/hub/models-adding-libraries.md:163]: Created a suggestion.

Comment (Wauplin) [reply docs/hub/models-adding-libraries.md]: Thanks! The VFI Mamba PR is indeed very nice with a complete PR and still concise. I've merged your suggestiong @Vaibhavs10 + updated the second link (Grok-1 PR was referenced twice). We should be good now!

**Pull request merged**